core/src/main/scala/org/apache/spark/util/UninterruptibleThread.scala

```diff
@@ -69,10 +75,22 @@ private[spark] class UninterruptibleThread(
     }
 
     uninterruptibleLock.synchronized {
+      uninterruptible = true
+    }
+
+    while (uninterruptibleLock.synchronized {
       // Clear the interrupted status if it's set.
       shouldInterruptThread = Thread.interrupted() || shouldInterruptThread
```

Comment:
is `Thread.interrupted()` a cheap function? Previously we only call it once, but now it's many times.

Reply:
shall we do it separately after this while loop?